hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/audit/TestAuditIntegration.java

```diff
@@ -67,6 +71,22 @@ public void testExceptionTranslation() throws Throwable {
         });
   }
 
+  /**
+   * UnsupportedRequest mapping and fail fast outcome.
+   */
+  @Test
+  public void testUnsupportedExceptionTranslation() throws Throwable {
+    final UnsupportedRequestException ex = intercept(UnsupportedRequestException.class, () -> {
+      throw translateException("test", "/",
+          new AuditOperationRejectedException("not supported"));
+    });
+    final S3ARetryPolicy retryPolicy = new S3ARetryPolicy(new Configuration(false));
+    final RetryPolicy.RetryAction action = retryPolicy.shouldRetry(ex, 1, 1, true);
```

Comment:
This should return "fail" request action with 0 for "retries" and "failovers", no?

```suggestion
    final RetryPolicy.RetryAction action = retryPolicy.shouldRetry(ex, 0, 0, true);
```

Reply:
will do